collect_app/src/main/java/org/odk/collect/android/instancemanagement/LocalInstancesUseCases.kt

```diff
@@ -0,0 +1,104 @@
+package org.odk.collect.android.instancemanagement
+
+import org.odk.collect.android.utilities.FileUtils
+import org.odk.collect.forms.instances.Instance
+import org.odk.collect.forms.instances.InstancesRepository
+import timber.log.Timber
+import java.io.File
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import java.util.TimeZone
+
+object LocalInstancesUseCases {
+    @JvmOverloads
+    @JvmStatic
+    fun createInstanceFileBasedOnFormPath(
+        formDefinitionPath: String,
+        instancesDir: String,
+        clock: () -> Long = { System.currentTimeMillis() }
+    ): File? {
+        val formFileName = formDefinitionPath
+            .substringAfterLast('/')
+            .substringBeforeLast('.')
+
+        return createInstanceFile(
+            formFileName,
+            instancesDir,
+            clock
+        )
+    }
+
+    fun clone(
+        instanceFile: File?,
+        instancesDir: String,
+        instancesRepository: InstancesRepository,
+        clock: () -> Long = { System.currentTimeMillis() }
+    ): Long? {
+        val sourceInstanceFile = instanceFile ?: return null
+        val targetInstanceFile = copyInstanceDir(sourceInstanceFile, instancesDir, clock) ?: return null
```

Comment:
Could we use the `displayName` from `Instance` to create the new file/directory instead of the original instance file name? I think would be simpler.

Reply:
> I could fetch it from the repository based on the instance file path, so it's doable. The function that generates the name would be slightly simpler, but it would break the convention - the displayName could end up being generated in any arbitrary way, rather than being based on the form name

Right I see: the instance file name is based on the blank form name, not the instance name. What I was trying to avoid is the logic that breaks apart the existing file name to create the new one which is sensitive to any change in how we generate file names. How would you feel about this grabbing the form name from `FormsRepository` instead from the file name?

Reply:
Hmm let me think how much changes that would require.

Reply:
Ok I did this: https://github.com/getodk/collect/pull/6646/commits/3623225b574ad67ce00b1229b3e618f219170f54